root/var/lib/vz/snippets/exec-cmds

```diff
@@ -27,6 +27,22 @@ exec_pci_rescan() {
   echo 1 > /sys/bus/pci/rescan
 }
 
+exec_set_haltpoll() {
+  echo "Setting haltpoll for $VMID..."
+  echo $1 > /sys/module/kvm/parameters/halt_poll_ns
+}
+
+exec_assign_interrupts() {
+  ##Wait 30s for all the interrupts to show up
+  sleep 30s
```

Comment:
I'm also curious about this, can you show what new interrupts popup, maybe less than 30s could be used?

Reply:
I tested smaller values but results were inconsistent. I think it is required for Windows to load up its GPU driver and only then all of the vfio-gpu interrupts start happening. On hardware slower than mine it might take longer, on hardware faster than mine it might be faster. 30 seconds is I think a generous value that gives enough time for Windows to start up. In my opinion it's better to wait and be of the safe side, otherwise not all interrupts are caught and moved and the optimization becomes useless. I am using irqtop to monitor the interrupts on my host machine and all other vfio interrupts show up immediately, only the GPU-related ones seem to take their time.